internal/venice-common/src/main/java/com/linkedin/venice/blobtransfer/NettyP2PBlobTransferManager.java

```diff
@@ -46,16 +49,19 @@ public CompletionStage<InputStream> get(String storeName, int version, int parti
     if (discoverPeers == null || discoverPeers.isEmpty()) {
       throw new VenicePeersNotFoundException("No peers found for the requested blob");
     }
-    try {
-      // TODO: add some retry logic or strategy to choose the peers differently in case of failure
-      // instanceName comes as a format of <hostName>_<applicationPort>
-      String chosenHost = discoverPeers.get(0).split("_")[0];
-      inputStream = nettyClient.get(chosenHost, storeName, version, partition);
-    } catch (Exception e) {
-      throw new VenicePeersNotFoundException("The connection to peers failed", e);
+    for (String peer: discoverPeers) {
+      try {
+        // TODO: add some retry logic or strategy to choose the peers differently in case of failure
+        // instanceName comes as a format of <hostName>_<applicationPort>
+        String chosenHost = peer.split("_")[0];
+        inputStream = nettyClient.get(chosenHost, storeName, version, partition);
+        LOGGER.info("Chosen host: {}", chosenHost);
+        return inputStream;
+      } catch (Exception e) {
+        throw new VenicePeersNotFoundException("The connection to peers failed", e);
```

Comment:
No so in this case, why are you still throwing Exception? We can just log the exception with WARN, and continue, until we find out the one that actually connects. Take a look at how this nettyClient.get() is implemented, you will see how it tries to establish connection to the host and what it will do if it fails..

Reply:
addressed in latest commit i believe